app/src/main/java/com/babylon/wallet/android/presentation/transaction/analysis/processor/PoolContributionProcessor.kt

```diff
@@ -22,31 +21,43 @@ import javax.inject.Inject
 
 class PoolContributionProcessor @Inject constructor(
     private val resolveAssetsFromAddressUseCase: ResolveAssetsFromAddressUseCase,
-    private val getPoolDetailsUseCase: GetPoolDetailsUseCase,
     private val getProfileUseCase: GetProfileUseCase
 ) : PreviewTypeProcessor<DetailedManifestClass.PoolContribution> {
-    @Suppress("LongMethod")
+
     override suspend fun process(summary: ExecutionSummary, classification: DetailedManifestClass.PoolContribution): PreviewType {
         val assets = resolveAssetsFromAddressUseCase(
             fungibleAddresses = summary.involvedFungibleAddresses(),
             nonFungibleIds = summary.involvedNonFungibleIds()
         ).getOrThrow()
-        val involvedPools = getPoolDetailsUseCase(classification.poolAddresses.map { it.addressString() }.toSet()).getOrThrow()
         val defaultDepositGuarantee = getProfileUseCase.invoke().first().appPreferences.transaction.defaultDepositGuarantee
         val accountsWithdrawnFrom = summary.accountWithdraws.keys
         val ownedAccountsWithdrawnFrom = getProfileUseCase.accountsOnCurrentNetwork().filter {
             accountsWithdrawnFrom.contains(it.address)
         }
         val from = summary.extractWithdraws(ownedAccountsWithdrawnFrom, assets)
-        val to = summary.accountDeposits.map { depositsPerAddress ->
+        val to = summary.extractDeposits(classification, assets, defaultDepositGuarantee)
+        return PreviewType.Transfer.Pool(
+            from = from,
+            to = to,
+            actionType = PreviewType.Transfer.Pool.ActionType.Contribution
+        )
+    }
+
+    private suspend fun ExecutionSummary.extractDeposits(
+        classification: DetailedManifestClass.PoolContribution,
+        assets: List<Asset>,
+        defaultDepositGuarantee: Double
+    ): List<AccountWithTransferableResources.Owned> {
+        val involvedPools = assets.filterIsInstance<PoolUnit>().mapNotNull { it.pool }
+        val to = accountDeposits.map { depositsPerAddress ->
             val ownedAccount = getProfileUseCase.accountOnCurrentNetwork(depositsPerAddress.key) ?: error("No account found")
             val deposits = depositsPerAddress.value.map { deposit ->
                 val resourceAddress = deposit.resourceAddress
                 val contributions = classification.poolContributions.filter {
                     it.poolUnitsResourceAddress.addressString() == resourceAddress
                 }
                 if (contributions.isEmpty()) {
-                    resolveGeneralAsset(deposit, summary, assets, defaultDepositGuarantee)
+                    resolveGeneralAsset(deposit, this, assets, defaultDepositGuarantee)
                 } else {
                     val pool = involvedPools.find { it.address == contributions.first().poolAddress.addressString() }
```

Comment:
Improvement for the future: 
These 6 lines could be also replaced if you didn't filter for involved pools only.

You could just find the pool unit by the resource address. Inside the pool unit object you have all you need.
* pool (with metadata and associated dapps)
* associated resources
* unit fungible

Reply:
makes sense. please verify if the recent change reflect what you had in mind